functions/daemon.go

```diff
@@ -244,6 +244,11 @@ func setupMQTT(server *config.Server) error {
 				nil,
 			)
 		}
+
+		// restart daemon for new udp hole punch if MQTT connection is lost (can happen on network change)
+		if !config.Netclient().IsStatic {
+			daemon.Restart()
```

Comment:
Should return here. There is no need for this goroutine to attempt connection if daemon is being restarted.

Reply:
that's a callback handler, it won't reach the code outside of it